scripts/build/gen_syscalls.py

```diff
@@ -424,6 +431,8 @@ def parse_args():
                         help="Generate marshalling files (*_mrsh.c)")
     parser.add_argument("-e", "--syscall-export-llext",
                         help="output C system call export for extensions")
+    parser.add_argument("-u", "--userspace-only", action="store_true",
+                        help="Only generate the userpace path of wrappers")
```

Comment:
It is - with the routing code, the `z_impl_*` will be undefined symbols on builds where only userspace matter. This can be a problem on some toolchains, so the application developer has to stub them. If the symbol is not there to start with, this issue is gone.